### ISSUE TITLE
implement cron scheduling

### ISSUE THREAD

**Issue by mheily** [mheily/jobd#17]
There is a public domain cron scheduling library called "libhoro" that looks like it would be easy to integrate with relaunchd:

http://gatanova.com:8081/cgi-bin/fossil.cgi/libhoro/doc/trunk/doc/faq.wiki

**Comment by davidchisnall** [comment]
There are three related features that are increasingly important for cron-style scheduling:
1. anachron-style scheduling, so that jobs are run even if the machine is powered down / asleep when they were initially scheduled.
2. Power awareness - jobs must be coalesced to minimise wakeup events and some should be deferred when not on mains power.  Similarly, the cron task is effectively a batch scheduler so should avoid thundering herds problems.

**Comment by mheily** [comment]
Good point; that essentially makes libhoro unsuitable, because it requires the program to wake up every minute, and will not reschedule jobs that were skipped while the machine was asleep or powered off.

I've implemented the manifest parser in b6b6ba2, and will write a new cron scheduler that:
- only wakes up when there is a job to be run
- will re-run jobs that were skipped for power-related reasons
- will not launch multiple instances of a job, to avoid the thundering herd
